### PR TITLE
94/t/STORY-3309/Migrate to Adwords api v201509

### PR DESCRIPTION
**Pull request by ibiziiac-sv**
WIP

### CODE REVIEW

**Comment by sherriman** [comment]
@ibiziiac-sv - What is the source of these changes?  Specifically, NoriLoader?  It isn't doing what you mean for it to do.  Specifically, it creates a class method and not an instance method.

**Comment by sherriman** [comment]
Never mind, I was using an old SHA when I was reviewing the code.  Note that I had to do `bundle update savon_zuora` to get the latest version of the gem installed.  `bundle install` when using a branch doesn't fetch the tip of the branch.  I didn't realize that.

**Comment by sherriman** [comment]
Okay, I get what is going on here now.

You are replacing `NoriSavon.configure` with just calling Nori.new with the correct parameters.  That looks better.

I assume the idea is not to change the global (shared) Nori configure, but rather use savon_zuora specific ones.  Is that correct?

Scott

**Comment by ibiziiac-sv** [comment]
I am intending to drop the forked `nori_savon` v.1.1.5 and use the up to date v2.4
I'm creating an isntance of Nori with NoriLoader because the new version uses instance methods, but the old version was using the class method `NoriSavon.parse`.

**Comment by ibiziiac-sv** [comment]
@sherriman please review

**Comment by codyjca** [comment]
@sherriman can you please take a look? I don't have a lot of context on the need for this change.

File: spec/support/fixture.rb [hunk 1]
```diff
@@ -31,5 +31,13 @@ def read_file(type, fixture)
       File.read path
     end
 
+    def nori
+      nori_options = {
```
Comment (sherriman) [inline spec/support/fixture.rb]: We should not evaluate nori_options every time you call this method.  You are creating objects you don't need, like a lambda

Move this into a method nori_options and call it here.  Look how simple this method gets.  And we only call nori_options as needed.

```
def nori
  @nori ||= Nori.new(nori_options)
end
```

**Comment by sherriman** [comment]
This looks good.  Comments given.

**Specific Comments**
Please address these:
- Can we afford to construct a new instance of nori on every soap response or fault?  That is, is the cost appropriate?
- What lead you to bump the gem dependencies in the gem spec?  For example,  Part of our confusion was changing HTTPI versions.  Did other changes require that?  How do we know this works with the new version of HTTPI, nori and wasabi?

**Comment by sherriman** [comment]
You should bump the version, right?

**Comment by ibiziiac-sv** [comment]
@sherriman I addressed the comments regarding `nori_options`.
Specific comments:
- With the new version of nori we have to create an instance to be able to use `parse` method, but the old version had the class method `parse`.
- The gems `nori`, `httpi` and `wasabi` are used by both `active_zuora` and `google-api-ads-ruby`. In order to keep google API client up to date, I decided to update these dependencies. But knowing that updating `active_zuora` and `savon` to newer versions would lead to massive changes to be applied in the `web` repository, I decided to adapt the existing compatibility versions, that were forked earlier and customized earlier, to work with the updated versions of `nori`, `httpi` and `wasabi`.

**Comment by sherriman** [comment]
:+1: 
Thank you for answering my comments.  Looks good.

**Comment by sherriman** [comment]
I missed this, we should probably bump the version.rb of this file as well.  Same as active_zuora.

Not merge in the latest gem, just change the version.rb.

**Comment by ibiziiac-sv** [comment]
But what version would it be? It's the same version as before, but customized with the updated version of `nori` usage.

**Comment by sherriman** [comment]
I think that 0.9.10 is correct.

**Pull request merged**